projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -77,7 +100,7 @@ function jetpack_add_google_fonts_provider() {
 	 *
 	 * @param array $fonts_to_register Array of Google Font names to register.
 	 */
-	$fonts_to_register = apply_filters( 'jetpack_google_fonts_list', JETPACK_GOOGLE_FONTS_LIST );
+	$fonts_to_register = apply_filters( 'jetpack_google_fonts_list', $fonts_list );
 
 	foreach ( $fonts_to_register as $font_family ) {
 		wp_register_webfonts(
```

Comment:
Frontend actually does support separate "name" as label instead of fontFamily which is used as the key, so there could be a way to inject different label from font slug.

https://github.com/WordPress/gutenberg/blob/f592820d07f1253251408b753ec5e575c2e79a52/packages/block-editor/src/components/font-family/index.js#L37

This is to have "Noto Sans JP" actually appear as "Noto Sans (Japanese)" in the font list.